### PR TITLE
tests: update the listing expression to support core from different channels

### PR DESCRIPTION
**Pull request by sergiocazzolato**

### CODE REVIEW

File: tests/main/listing/task.yaml [hunk 1]
```diff
@@ -20,19 +20,19 @@ execute: |
     #shellcheck disable=SC2166
     if [ "$SPREAD_BACKEND" = "linode" -o "$SPREAD_BACKEND" = "google" -o "$SPREAD_BACKEND" == "qemu" ] && [ "$SPREAD_SYSTEM" = "ubuntu-core-16-64" ]; then
         echo "With customized images the core snap is sideloaded"
-        expected='^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +x[0-9]+ +- +- +core *$'
+        expected="^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +x[0-9]+ +- +- +core *$"
```
Comment (zyga) [inline tests/main/listing/task.yaml]: What's the point of changing the string to double quotes? Is this a shellcheck-driven change?

File: tests/main/listing/task.yaml [hunk 2]
```diff
@@ -20,19 +20,19 @@ execute: |
...
     elif [ "$SRU_VALIDATION" = "1" ]; then
         echo "When sru validation is done the core snap is installed from the store"
-        expected='^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+[0-9]+\.[0-9a-f]+)? +[0-9]+ +stable +canonical✓ +core *$'
+        expected="^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+[0-9]+\.[0-9a-f]+)? +[0-9]+ +stable +canonical✓ +core *$"
     elif [ "$SPREAD_BACKEND" = "external" ] || [ "$SPREAD_BACKEND" = "autopkgtest" ]; then
-        expected='^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +[0-9]+ +(edge|beta|candidate|stable) +canonical✓ +core *$'
+        expected="^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +[0-9]+ +(edge|beta|candidate|stable) +canonical✓ +core *$"
     else
-        expected='^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +[0-9]+ +edge +canonical✓ +core *$'
+        expected="^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +[0-9]+ +$CORE_CHANNEL +canonical✓ +core *$"
```
Comment (zyga) [inline tests/main/listing/task.yaml]: This perhaps looks fine except that I would change all `\` characters to `\\` along with the change to double-quoted strings.

**Comment by zyga** [comment]
There's a lot of errors from shellcheck, perhaps those are due to merging of older branch with master?

```
ERROR:root:shellcheck failed for file tests/unit/c-unit-tests-gcc/task.yaml in sections: execute; error log follows
ERROR:root:tests/unit/c-unit-tests-gcc/task.yaml: section 'execute':

In - line 1:
cd "$SPREAD_PATH/cmd/"
^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.


In - line 4:
cd $build_dir
^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
   ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/unit/c-unit-tests-clang/task.yaml in sections: execute; error log follows
ERROR:root:tests/unit/c-unit-tests-clang/task.yaml: section 'execute':

In - line 1:
cd "$SPREAD_PATH/cmd/"
^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.


In - line 4:
cd $build_dir
^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
   ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/regression/lp-1644439/task.yaml in sections: prepare, execute; error log follows
ERROR:root:tests/regression/lp-1644439/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:tests/regression/lp-1644439/task.yaml: section 'execute':

In - line 21:
. "$TESTSLIB/dirs.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1595444/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1595444/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1615113/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1615113/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1667385/task.yaml in sections: execute; error log follows
ERROR:root:tests/regression/lp-1667385/task.yaml: section 'execute':

In - line 2:
snap list $SNAP | MATCH "$FLAG$"
          ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/regression/lp-1665004/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1665004/task.yaml: section 'prepare':

In - line 1:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1630479/task.yaml in sections: prepare, execute; error log follows
ERROR:root:tests/regression/lp-1630479/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:tests/regression/lp-1630479/task.yaml: section 'execute':

In - line 1:
. "$TESTSLIB/dirs.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In - line 3:
revision=$(readlink "$SNAP_MOUNT_DIR/test-snapd-tools/current")
^-- SC2034: revision appears unused. Verify use (or export if used externally).


In - line 4:
one="$(test-snapd-tools.cmd sh -c 'echo $PATH')"
                                  ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.


In - line 5:
two="$(test-snapd-tools.cmd sh -c 'echo $PATH')"
                                  ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.


ERROR:root:shellcheck failed for file tests/regression/lp-1613845/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1613845/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1618683/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1618683/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1597839/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1597839/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1764977/task.yaml in sections: execute; error log follows
ERROR:root:tests/regression/lp-1764977/task.yaml: section 'execute':

In - line 1:
cp -a  $TESTSLIB/snaps/test-snapd-sh .
       ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/regression/lp-1597842/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1597842/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1599891/task.yaml in sections: execute; error log follows
ERROR:root:tests/regression/lp-1599891/task.yaml: section 'execute':

In - line 5:
MATCH "$snap_confine \(enforce\)" < /sys/kernel/security/apparmor/profiles
                     ^-- SC1117: Backslash is literal in "\(". Prefer explicit escaping: "\\(".
                              ^-- SC1117: Backslash is literal in "\)". Prefer explicit escaping: "\\)".


ERROR:root:shellcheck failed for file tests/regression/lp-1606277/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1606277/task.yaml: section 'prepare':

In - line 1:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1732555/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1732555/task.yaml: section 'prepare':

In - line 1:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/regression/lp-1704860/task.yaml in sections: execute; error log follows
ERROR:root:tests/regression/lp-1704860/task.yaml: section 'execute':

In - line 1:
. $TESTSLIB/snaps.sh
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
  ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/regression/lp-1607796/task.yaml in sections: prepare; error log follows
ERROR:root:tests/regression/lp-1607796/task.yaml: section 'prepare':

In - line 2:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:shellcheck failed for file tests/main/ubuntu-core-writablepaths/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/ubuntu-core-writablepaths/task.yaml: section 'execute':

In - line 2:
cat /etc/system-image/writable-paths | while read -r line; do
    ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In - line 3:
    line=$(echo $line | sed -e '/\s*#.*$/d')
                ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 8:
    dir_or_file=$(echo $line|cut -f1 -d' ')
                       ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 18:
    rm -f $dir_or_file/random-name-that-I-made-up
          ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/main/snapctl-configure-core/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/snapctl-configure-core/task.yaml: section 'execute':

In - line 10:
cat /etc/systemd/logind.conf.d/00-snap-core.conf | MATCH HandlePowerKey=reboot
    ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In - line 24:
    cat /boot/uboot/config.txt | MATCH ^hdmi_mode=1
        ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In - line 26:
    cat /boot/uboot/config.txt | MATCH ^#hdmi_mode=1
        ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In - line 38:
    cat /etc/environment | MATCH "^${proto}_proxy=http://example.com:8021"
        ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In - line 40:
    if cat /etc/environment | grep "^{proto}_proxy="; then
           ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In - line 49:
cat /etc/environment | MATCH "^no_proxy=example.com,bar.com"
    ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In - line 51:
if cat /etc/environment | grep "^no_proxy="; then
       ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


ERROR:root:shellcheck failed for file tests/main/try-twice-with-daemon/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/try-twice-with-daemon/task.yaml: section 'execute':

In - line 3:
snap try $TESTSLIB/snaps/test-snapd-service-try-v1
         ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 12:
if systemctl|grep -q $SERVICE ; then
                     ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 19:
snap try $TESTSLIB/snaps/test-snapd-service-try-v2
         ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 24:
output=$(systemctl show --property=LoadState $SERVICE)
                                             ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/main/ubuntu-core-services/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/ubuntu-core-services/task.yaml: section 'execute':

In - line 10:
for timer in snapd.snap-repair.timer; do
             ^-- SC2043: This loop will only ever run once for a constant value. Did you perhaps mean to loop over dir/*, $var or $(cmd)?


ERROR:root:shellcheck failed for file tests/main/snap-userd-reexec/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/snap-userd-reexec/task.yaml: section 'execute':

In - line 7:
snap install --dangerous /var/lib/snapd/snaps/core_$(cat prevBoot).snap
                                                   ^-- SC2046: Quote this to prevent word splitting.


ERROR:root:shellcheck failed for file tests/main/snap-set/task.yaml in sections: prepare; error log follows
ERROR:root:tests/main/snap-set/task.yaml: section 'prepare':

In - line 8:
snap pack $TESTSLIB/snaps/failing-config-hooks
          ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/main/snap-service-watchdog/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/snap-service-watchdog/task.yaml: section 'execute':

In - line 2:
. $TESTSLIB/snaps.sh
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
  ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 21:
    test "$wdk" -ge 4 && break || true
                      ^-- SC2015: Note that A && B || C is not if-then-else. C may run when A is true.


In - line 34:
for service in direct-watchdog-bad; do
               ^-- SC2043: This loop will only ever run once for a constant value. Did you perhaps mean to loop over dir/*, $var or $(cmd)?


In - line 36:
        systemctl show -p SubState snap.test-snapd-service-watchdog.$service | MATCH 'SubState=(failed|stop-sigabrt)' && break || true
                                                                                                                      ^-- SC2015: Note that A && B || C is not if-then-else. C may run when A is true.


In - line 39:
    systemctl show -p SubState snap.test-snapd-service-watchdog.$service | MATCH 'SubState=(failed|stop-sigabrt)' && break || true
                                                                                                                  ^-- SC2015: Note that A && B || C is not if-then-else. C may run when A is true.


ERROR:root:shellcheck failed for file tests/main/user-mounts/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/user-mounts/task.yaml: section 'execute':

In - line 2:
. $TESTSLIB/dirs.sh
  ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 6:
mkdir -p $USER_RUNTIME_DIR || true
         ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 8:
rm -rf $USER_RUNTIME_DIR/*
       ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 9:
chmod u=rwX,go= $USER_RUNTIME_DIR
                ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 10:
chown test:test $USER_RUNTIME_DIR
                ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/main/ubuntu-core-gadget-config-defaults/task.yaml in sections: prepare, restore; error log follows
ERROR:root:tests/main/ubuntu-core-gadget-config-defaults/task.yaml: section 'prepare':

In - line 5:
. $TESTSLIB/systemd.sh
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
  ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 37:
cp $TESTSLIB/assertions/developer1.account /var/lib/snapd/seed/assertions
   ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 38:
cp $TESTSLIB/assertions/developer1.account-key /var/lib/snapd/seed/assertions
   ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 39:
cp $TESTSLIB/assertions/developer1-pc-w-config.model /var/lib/snapd/seed/assertions
   ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 40:
cp $TESTSLIB/assertions/testrootorg-store.account-key /var/lib/snapd/seed/assertions
   ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:tests/main/ubuntu-core-gadget-config-defaults/task.yaml: section 'restore':

In - line 10:
. $TESTSLIB/systemd.sh
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
  ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 26:
if systemctl status snap-test-snapd-with-configure-${TEST_REVNO}.mount ; then
                                                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 27:
   systemctl stop snap-test-snapd-with-configure-${TEST_REVNO}.mount
                                                 ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 28:
   rm -f /etc/systemd/system/snap-test-snapd-with-configure-${TEST_REVNO}.mount
                                                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 29:
   rm -f /etc/systemd/system/multi-user.target.wants/snap-test-snapd-with-configure-${TEST_REVNO}.mount
                                                                                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 42:
rm -f *.bak
      ^-- SC2035: Use ./*glob* or -- *glob* so names with dashes won't become options.


ERROR:root:shellcheck failed for file tests/main/systemd-service/task.yaml in sections: restore, execute; error log follows
ERROR:root:tests/main/systemd-service/task.yaml: section 'restore':

In - line 1:
rm -f *.snap
      ^-- SC2035: Use ./*glob* or -- *glob* so names with dashes won't become options.


ERROR:root:tests/main/systemd-service/task.yaml: section 'execute':

In - line 1:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In - line 7:
while ! systemctl show -p ActiveState $SERVICE_NAME | grep -Pq "ActiveState=active"; do sleep 0.5; done
                                      ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 10:
expected="(?s).*?Loaded: loaded .*?Active: active \(running\)"
                                                  ^-- SC1117: Backslash is literal in "\(". Prefer explicit escaping: "\\(".
                                                           ^-- SC1117: Backslash is literal in "\)". Prefer explicit escaping: "\\)".


In - line 11:
systemctl status $SERVICE_NAME | grep -Pqz "$expected"
                 ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/main/validate-container-failures/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/validate-container-failures/task.yaml: section 'execute':

In - line 5:
snap try $TESTSLIB/snaps/$SNAP 2>error.log && exit 1 || true
         ^-- SC2086: Double quote to prevent globbing and word splitting.
                         ^-- SC2086: Double quote to prevent globbing and word splitting.
                                           ^-- SC2015: Note that A && B || C is not if-then-else. C may run when A is true.


In - line 7:
tr -s "\n " " " < error.log | MATCH 'contact developer'
       ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


ERROR:root:shellcheck failed for file tests/main/ubuntu-core-custom-device-reg-extras/task.yaml in sections: prepare, restore; error log follows
ERROR:root:tests/main/ubuntu-core-custom-device-reg-extras/task.yaml: section 'prepare':

In - line 5:
. $TESTSLIB/systemd.sh
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
  ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 19:
cp $TESTSLIB/assertions/developer1.account /var/lib/snapd/seed/assertions
   ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 20:
cp $TESTSLIB/assertions/developer1.account-key /var/lib/snapd/seed/assertions
   ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 21:
cp $TESTSLIB/assertions/developer1-pc.model /var/lib/snapd/seed/assertions
   ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 22:
cp $TESTSLIB/assertions/testrootorg-store.account-key /var/lib/snapd/seed/assertions
   ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 24:
systemd_create_and_start_unit fakedevicesvc "$(which fakedevicesvc) localhost:11029"
                                               ^-- SC2230: which is non-standard. Use builtin 'command -v' instead.


ERROR:root:tests/main/ubuntu-core-custom-device-reg-extras/task.yaml: section 'restore':

In - line 5:
. $TESTSLIB/systemd.sh
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
  ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 25:
rm -f *.bak
      ^-- SC2035: Use ./*glob* or -- *glob* so names with dashes won't become options.


ERROR:root:shellcheck failed for file tests/main/listing/task.yaml in sections: execute; error log follows
ERROR:root:tests/main/listing/task.yaml: section 'execute':

In - line 8:
    expected="^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +x[0-9]+ +- +- +core *$"
                                                     ^-- SC1117: Backslash is literal in "\+". Prefer explicit escaping: "\\+".
                                                                ^-- SC1117: Backslash is literal in "\.". Prefer explicit escaping: "\\.".


In - line 11:
    expected="^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+[0-9]+\.[0-9a-f]+)? +[0-9]+ +stable +canonicalâ +core *$"
                                                     ^-- SC1117: Backslash is literal in "\+". Prefer explicit escaping: "\\+".
                                                             ^-- SC1117: Backslash is literal in "\.". Prefer explicit escaping: "\\.".


In - line 13:
    expected="^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +[0-9]+ +(edge|beta|candidate|stable) +canonicalâ +core *$"
                                                     ^-- SC1117: Backslash is literal in "\+". Prefer explicit escaping: "\\+".
                                                                ^-- SC1117: Backslash is literal in "\.". Prefer explicit escaping: "\\.".


In - line 15:
    expected="^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +[0-9]+ +$CORE_CHANNEL +canonicalâ +core *$"
                                                     ^-- SC1117: Backslash is literal in "\+". Prefer explicit escaping: "\\+".
                                                                ^-- SC1117: Backslash is literal in "\.". Prefer explicit escaping: "\\.".


In - line 20:
snap list | MATCH "^test-snapd-tools +[0-9]+(\.[0-9]+)* +x[0-9]+ +- +- +- *$"
                                             ^-- SC1117: Backslash is literal in "\.". Prefer explicit escaping: "\\.".


ERROR:root:shellcheck failed for file tests/main/xdg-settings/task.yaml in sections: restore, execute; error log follows
ERROR:root:tests/main/xdg-settings/task.yaml: section 'restore':

In - line 1:
. "$TESTSLIB/dirs.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In - line 2:
. "$TESTSLIB/pkgdb.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


ERROR:root:tests/main/xdg-settings/task.yaml: section 'execute':

In - line 1:
. "$TESTSLIB/pkgdb.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In - line 2:
. "$TESTSLIB/dirs.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In - line 3:
. "$TESTSLIB/snaps.sh"
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In - line 9:
export $(cat dbus.env | xargs)
       ^-- SC2046: Quote this to prevent word splitting.
             ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In - line 54:
! $SNAP_MOUNT_DIR/core/current/usr/bin/xdg-settings set random-settting something
  ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 56:
! $SNAP_MOUNT_DIR/core/current/usr/bin/xdg-settings set default-web-browser inĂ¤lid
  ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/main/snapctl-services/task.yaml in sections: restore, execute; error log follows
ERROR:root:tests/main/snapctl-services/task.yaml: section 'restore':

In - line 1:
rm -f $SERVICEOPTIONFILE
      ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:tests/main/snapctl-services/task.yaml: section 'execute':

In - line 5:
    while ! snap services $1 | MATCH $2; do
                          ^-- SC2086: Double quote to prevent globbing and word splitting.
                                     ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 17:
    while ! cat $1 | MATCH $2; do
                ^-- SC2086: Double quote to prevent globbing and word splitting.
                ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
                           ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 66:
wait_for_file_change $SERVICEOPTIONFILE "^foo$"
                     ^-- SC2086: Double quote to prevent globbing and word splitting.


ERROR:root:shellcheck failed for file tests/main/stale-base-snap/task.yaml in sections: prepare, execute; error log follows
ERROR:root:tests/main/stale-base-snap/task.yaml: section 'prepare':

In - line 3:
. $TESTSLIB/snaps.sh
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
  ^-- SC2086: Double quote to prevent globbing and word splitting.


In - line 8:
unsquashfs -no-progress $(ls /var/lib/snapd/snaps/core_*.snap | sort -r -n | head -n 1)
                        ^-- SC2046: Quote this to prevent word splitting.
                          ^-- SC2012: Use find instead of ls to better handle non-alphanumeric filenames.


ERROR:root:tests/main/stale-base-snap/task.yaml: section 'execute':

In - line 2:
test-snapd-sh -c 'touch $SNAP_DATA/stamp && exec sleep 1h' &
                 ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.


In - line 7:
for i in $(seq 30); do
^-- SC2034: i appears unused. Verify use (or export if used externally).

```

File: tests/main/listing/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/main/listing/task.yaml]: @zyga Thanks for reviewing it
You are right, no need to change that but I did it just to use the same type of quoting in all the lines. 
I am reverting that.

File: tests/main/listing/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/main/listing/task.yaml]: Change done

**Pull request merged**